src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/MissingAnnotationsDiagnosticRule.java

```diff
@@ -7,61 +7,60 @@
+                            "https://azure.github.io/azure-sdk/java_introduction.html#service-client-creation"));
                     }
-                });
+                } else if (className.endsWith("Client")) {
+                    // check if the @ServiceClient annotation is present
+                    if (!typeDeclaration.isAnnotationPresent("ServiceClient")) {
+                        listing.addDiagnostic(new Diagnostic(
+                            INFO,
+                            makeId(typeDeclaration),
+                            "Classes named *Client are potential candidates to have the @ServiceClient annotation applied.",
+                            "https://azure.github.io/azure-sdk/java_introduction.html#service-client"));
+                    }
+
+                    // check all public / protected methods in client classes. Because we can't easily determine if a method
+                    // should have an annotation, all we can do is count the number of methods that are annotated and compare
+                    // this with the total number of methods. If the ratio is not high enough, we will warn the user that there
+                    // may be missing annotations.
+                    final AtomicInteger methodCount = new AtomicInteger();
+                    final AtomicInteger annotatedMethodCount = new AtomicInteger();
+                    getPublicOrProtectedMethods(cu).forEach(methodDeclaration -> {
+                        methodCount.incrementAndGet();
+                        if (methodDeclaration.isAnnotationPresent("ServiceMethod")) {
+                            annotatedMethodCount.incrementAndGet();
+                        }
+                    });
 
-                if (annotatedMethodCount.get() / (double) methodCount.get() < 0.75) {
-                    // warn user to double check
-                    listing.addDiagnostic(new Diagnostic(
-                        WARNING,
-                        makeId(cu),
-                    "There is a low number of methods annotated with @ServiceMethod. " +
-                            "Please review to ensure all appropriate methods have this annotation.",
-                        "https://azure.github.io/azure-sdk/java_introduction.html#service-client"));
+                    if (annotatedMethodCount.get() / (double) methodCount.get() < 0.75) {
+                        // warn user to double check
+                        listing.addDiagnostic(new Diagnostic(
+                            WARNING,
+                            makeId(typeDeclaration),
+                        "There is a low number of methods annotated with @ServiceMethod. " +
+                                "Please review to ensure all appropriate methods have this annotation.",
+                            "https://azure.github.io/azure-sdk/java_introduction.html#service-client"));
+                    }
```

Comment:
FYI - the bulk of this change is simply an indentation by one level. The only notable change is the line which has the `.filter(type -> getPackageName(type).equals("com.azure"))` on it.